welkin/models/base.py

```diff
@@ -20,7 +20,15 @@ def __getattr__(self, name):
 
 
 class Resource(dict, SchemaBase):
+    nested_objects = {}
+
     def __getattr__(self, name):
+        if name in self.nested_objects:
+            cls = self.nested_objects[name]
```

Comment:
We can simplify this a little by capturing the value first and sorting out the rest later:

```python
def __getattr__(self, name):
        try:
            value = super().__getitem__(name)
        except KeyError:
            value = super().__getattr__(name)

        if name in self.nested_objects:
            return self.nested_objects[name](value)

        return value
```

Reply:
Nice